File: apps/ios/SofaExchange/Presentation/Search/SearchViewModel.swift
import Foundation

@MainActor
final class SearchViewModel: ObservableObject {
    var selectedCities: [City]      = []
    var hasFreeWifi: Bool?           = nil
    var selectedSofaType: SofaType?  = nil

    func buildQuery() -> SearchQuery {
        SearchQuery(
            cities:      selectedCities,
            hasFreeWifi: hasFreeWifi,
            sofaType:    selectedSofaType
        )
    }
}

File: apps/android/app/src/main/java/com/sofaexchange/presentation/search/SearchViewModel.kt
package com.sofaexchange.presentation.search

import androidx.lifecycle.ViewModel
import com.sofaexchange.domain.model.City
import com.sofaexchange.domain.model.SofaType
import com.sofaexchange.domain.repository.SearchParams

class SearchViewModel : ViewModel() {
    var selectedCities: MutableList<City> = mutableListOf()
    var hasFreeWifi: Boolean? = null
    var selectedSofaType: SofaType? = null

    fun buildSearchParams() = SearchParams(
        cities      = selectedCities.toList(),
        hasFreeWifi = hasFreeWifi,
        sofaType    = selectedSofaType,
    )
}
